projectgenerator/libili2pg/ilicache.py

```diff
@@ -130,29 +132,48 @@ def process_local_ili_folder(self, path):
         '''
         Parses all .ili files in the given ``path`` (non-recursively)
         '''
-        model = dict()
         models = list()
-        re_model = re.compile('\s*MODEL\s*([\w\d_-]+)\s.*')
-        re_model_version = re.compile('VERSION "([ \w\d\._-]+)".*')
-
+        fileModels = list()
         for ilifile in glob.iglob(os.path.join(path, '*.ili')):
-            with open(ilifile, 'r') as file:
-                for line in file:
-                    result = re_model.search(line)
-                    if result:
-                        model = dict()
-                        model['name'] = result.group(1)
-                        model['version'] = ''
-                        models += [model]
-
-                    result = re_model_version.search(line)
-                    if result:
-                        model['version'] = result.group(1)
-
-        print(repr(models[0]['name']))
+            try:
+                fileModels = self.parse_ili_file(ilifile, "utf-8")
+            except UnicodeDecodeError:
+                try:
+                    fileModels = self.parse_ili_file(ilifile, "latin1")
+                    self.iface.messageBar().pushWarning(self.tr('Project Generator'),
```

Comment:
Hmm, I think it would be good to remove the dependency on `iface` within this module. The reason is that it should also be used from a standalone app / travis ci script without the whole QGIS application running behind and a different error reporting approach. The easiest is probably to switch this over to a signal that is emitted from this class and let the `GenerateProjectDialog` handle sending this to the message bar.